tests/integrations/examples/discrete_nolog_train.py

```diff
@@ -0,0 +1,93 @@
+"""test runner for Hubbard example."""
```

Comment:
Similarly, what's the difference between this and the sgd runner? Seems to me that a lot of logic is duplicated here, probably better to add desired functionality to the already existing runners?